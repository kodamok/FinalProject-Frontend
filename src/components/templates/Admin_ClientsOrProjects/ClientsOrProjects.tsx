import React, { useContext, useEffect, useState } from 'react';
import { BsThreeDots } from 'react-icons/bs';
import styled from 'styled-components';
import IconClickable from '../../atoms/IconClickable/IconClickable';
import Button from '../../atoms/Button/Button';
import InputWithLabel from '../../atoms/InputWithLabel/InputWithLabel';
import { Context } from '../../../providers/GeneralProvider';
import CardClient from '../../molecules/CardClient/CardClient';
import CardProject from '../../molecules/CardProject/CardProject';
import useError from '../../../hooks/useError';
import NoItemsFound from '../../atoms/NoItemsFound/NoItemsFound';

const ContainerFilterBy = styled.div`
  display: flex;
  justify-content: space-between;
  padding: 0 1rem;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  > div:first-child {
    display: flex;
    align-items: center;
  }
  ${({ theme }) => theme.up(theme.breakpoint.m)} {
    padding: 0 2rem;
  }
`;

const ContainerClients = styled.div`
  padding: 0.3rem;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  gap: 1rem;
  max-width: 95%;
  margin: 1.2rem auto 1.2rem auto;
  border-radius: 1rem;
  ${({ theme }) => theme.down('700px')} {
    grid-template-columns: repeat(auto-fit, minmax(300px, 400px));
    justify-content: center;
  }
`;
const ContainerProjects = styled.div`
  padding: 0.3rem;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
  gap: 1rem;
  max-width: 95%;
  margin: 1.2rem auto 1.2rem auto;
  border-radius: 1rem;
  ${({ theme }) => theme.down('700px')} {
    grid-template-columns: repeat(auto-fit, minmax(300px, 400px));
    justify-content: center;
  }
`;

function ClientsOrProjects() {
  const [choiceRadio, setChoiceRadio] = useState('projects');
  const [clients, setClients] = useState([]);
  console.log(clients[0]);

  const [projects, setProjects] = useState([]);
  const { userData } = useContext(Context);
  const { handleError } = useError();

  const fetchClients = async () => {
    try {
      const res = await fetch(`${process.env.REACT_APP_BACKEND}/user/freelancer`, {
        method: 'GET',
        headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${userData.token}`
        }
      });
      const resJSON = await res.json();
      if (res.status === 200) {
        setClients(resJSON);
      } else {
        handleError(resJSON.message);
      }
    } catch (error: any) {
      console.log('FETCHING ERROR', error);
      handleError();
    }
  };

  const fetchProjects = async () => {
    try {
      const res = await fetch(`${process.env.REACT_APP_BACKEND}/project`, {
        method: 'GET',
        headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${userData.token}`
        }
      });
      const resJSON = await res.json();
      console.log(resJSON);
      if (res.status === 200) {
        setProjects(resJSON);
      } else {
        handleError(resJSON.message);
      }
    } catch (error: any) {
      console.log('FETCHING ERROR', error);
      handleError();
    }
  };

  useEffect(() => {
    fetchClients();
    fetchProjects();
  }, []);

  console.log({ clients, projects });

  // Handling value when you click on choice
  const handleChangeRadio = (e: any) => {
    setChoiceRadio(e.target.value);
  };
  console.log(choiceRadio);
  return (
    <div>
      <h2>Clients/Projects</h2>
      <ContainerFilterBy>
        <div>
          <p>Filter by:</p>
          <InputWithLabel
            type="radio"
            label="Projects"
            name="chooseType"
            onChange={handleChangeRadio}
            value="projects"
            id="projects"
            checked={choiceRadio === 'projects'}
          />
          <InputWithLabel
            type="radio"
            label="Clients"
            name="chooseType"
            value="clients"
            id="clients"
            onChange={handleChangeRadio}
            checked={choiceRadio === 'clients'}
          />
        </div>
        <div>
          <IconClickable icon={<BsThreeDots fontSize={28} />}>
            <div
              style={{ display: 'flex', flexDirection: 'column', gap: '0.5rem', padding: '1rem' }}
            >
              <Button
                text="Create New Client"
                width="180px"
                fontSize="1rem"
                padding="0.5rem 1rem"
              />
              <Button
                text="Create New Project"
                width="180px"
                fontSize="1rem"
                padding="0.5rem 1rem"
              />
            </div>
          </IconClickable>
        </div>
      </ContainerFilterBy>
      {choiceRadio === 'projects' ? (
        <ContainerProjects>
          {projects.length ? (
            projects.map((item: any) => <CardProject key={item._id} projectData={item} />)
          ) : (
            <NoItemsFound text="Projects" />
          )}
        </ContainerProjects>
      ) : (
        <ContainerClients>
          {clients.length ? (
            clients.map((item: any) => <CardClient key={item._id} clientData={item} />)
          ) : (
            <NoItemsFound text="Clients" />
          )}
        </ContainerClients>
      )}
    </div>
  );
}

export default ClientsOrProjects;
