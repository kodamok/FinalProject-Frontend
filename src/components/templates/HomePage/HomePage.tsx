import React from 'react';
import Header from '../../organisms/Header/Header';
import HeroSection from '../../organisms/HeroSection/HeroSection';
import SectionProductDesc1 from '../../organisms/SectionProductDesc1/SectionProductDesc1';
import SectionProductDesc2 from '../../organisms/SectionProductDesc2/SectionProductDesc2';
import SectionProductDesc3 from '../../organisms/SectionProductDesc3/SectionProductDesc3';

function HomePage() {
  return (
    <div>
      <Header />
      <HeroSection />
      <SectionProductDesc1 />
      <SectionProductDesc2 />
      <SectionProductDesc3 />
    </div>
  );
}

export default HomePage;
