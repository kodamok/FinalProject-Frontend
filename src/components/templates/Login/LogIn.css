.login{
    padding:0 3rem;   
}
h3{
    text-align: center
}
p{
    text-align:right;
}

.btn{
    display:flex;
    flex-direction: column;
    align-items:center;
    gap:1rem;

}

.line{
    width:50%;
    height: 2px;
    background-color:#666;
    /* margin-top: 10px;
    margin-bottom: 15px; */
    
}