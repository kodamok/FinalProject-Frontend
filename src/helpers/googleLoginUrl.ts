import * as queryString from 'query-string';

const stringifiedParams = queryString.stringify({
  client_id: process.env.REACT_APP_GOOGLE_CLIENT_ID,
  redirect_uri: process.env.REACT_APP_FRONTEND,
  scope: [
    'https://www.googleapis.com/auth/userinfo.email',
    'https://www.googleapis.com/auth/userinfo.profile'
  ].join(' '), // space seperated string
  response_type: 'code',
  access_type: 'offline',
  prompt: 'consent'
});

export const googleLoginUrl = `https://accounts.google.com/o/oauth2/v2/auth?${stringifiedParams}`;
