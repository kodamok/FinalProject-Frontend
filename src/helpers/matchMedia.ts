// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.matchMedia = (query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(), // Deprecated
  removeListener: jest.fn(), // Deprecated
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn()
});
