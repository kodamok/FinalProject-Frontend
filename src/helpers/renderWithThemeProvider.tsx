import * as React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter } from 'react-router-dom';
import { theme } from '../assets/styles/theme';
import { GlobalStyle } from '../assets/styles/GlobalStyle';
import MainContainerApp from '../components/templates/MainContainer/MainContainer';
import GeneralProvider from '../providers/GeneralProvider';

export const renderWithThemeProvider = (elem: any) =>
  render(
    <GeneralProvider>
      <BrowserRouter>
        <ThemeProvider theme={theme}>
          <GlobalStyle />
          <MainContainerApp>{elem}</MainContainerApp>
        </ThemeProvider>
      </BrowserRouter>
    </GeneralProvider>
  );
