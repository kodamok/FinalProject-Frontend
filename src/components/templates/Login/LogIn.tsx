import React, { SyntheticEvent, useContext } from 'react';
import styled from 'styled-components';
import { useNavigate } from 'react-router-dom';
import { FcGoogle } from 'react-icons/fc';
import InputWithLabel from '../../atoms/InputWithLabel/InputWithLabel';
import Button from '../../atoms/Button/Button';
import './LogIn.css';
import useForm from '../../../hooks/useForm';
import { Context } from '../../../providers/GeneralProvider';

const Container = styled.form`
  padding: 0 1rem;
  max-width: 350px;
  margin: 0 auto;

  h3 {
    text-align: center;
  }
  p {
    margin: 0.4rem 0 1rem 0;
    text-align: right;
    display: inline-block;
  }
  p:hover {
    cursor: pointer;
    text-decoration: underline;
  }
  > div:last-child {
    display: flex;
    flex-direction: column;
    align-items: center;
    gap: 1rem;
  }

  ${({ theme }) => theme.up(theme.breakpoint.m)} {
    max-width: 350px;
  }
`;

const ContainerP = styled.div`
  display: flex;
  justify-content: flex-end;
`;

const Line = styled.div`
  width: 230px;
  height: 2px;
  background-color: #666;
`;

function LogIn() {
  const { setUserData } = useContext(Context);
  interface FormLogin {
    email: string;
    password: string;
  }
  const navigate = useNavigate();

  const initialValue: FormLogin = {
    email: '',
    password: ''
  };

  const { handleChange, inputs } = useForm(initialValue);

  const handleSubmit = async (e: SyntheticEvent) => {
    e.preventDefault();
    console.log('You try login with these inputs', inputs);
    const login = async () => {
      try {
        const res = await fetch(`${process.env.REACT_APP_BACKEND}/login`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json'
            // Authorization: 'Bearer ' + userData?.token, // IN FUTURE TO AUTHORIZATION
          },
          body: JSON.stringify(inputs)
        });
        const resJSON = await res.json();
        console.log(resJSON);
        if (res.status === 200) {
          setUserData(resJSON);
          navigate('/');
        }
      } catch (error: any) {
        console.log('FETCHING ERROR', error);
      }
    };
    login();
  };

  const handleNavigateToForgotPassword = () => {
    navigate('/forgotPassword');
  };

  return (
    <Container onSubmit={handleSubmit}>
      <h3>LOGIN</h3>
      <InputWithLabel label="Email" name="email" onChange={handleChange} />
      <InputWithLabel type="password" label="Password" name="password" onChange={handleChange} />
      <ContainerP>
        <p onClick={handleNavigateToForgotPassword}>I forgot my password</p>
      </ContainerP>
      <div>
        <Button type="submit" background="#2A9D8F" text="Login" />
        <Line />
        <Button text="Login with Gmail" icon={<FcGoogle />} />
        <Button background="#9e0059" text="Create New Account" />
      </div>
    </Container>
    // <div className="login">
    //   <h3 className="h3">LOGIN</h3>
    //   <InputWithLabel label="Email" name="email" />
    //   <InputWithLabel label="Password" name="password" />
    //   <p className="p">I forgot my password</p>
    //   <div className="btn">
    //     <Button background="#2A9D8F" text="Login" />
    //     <div className="line" />
    //     <Button text="Login with Gmail" icon={<FcGoogle />} />
    //     <Button background="#9e0059" text="Create New Account" />
    //   </div>
    // </div>
  );
}

export default LogIn;
