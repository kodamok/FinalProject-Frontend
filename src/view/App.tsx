import React, { useContext, useEffect, useState } from 'react';
import { Routes, Route, useNavigate } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import * as queryString from 'query-string';
import HomePage from '../components/templates/HomePage/HomePage';
import Services from '../components/templates/Services/Services';
import SignUp from '../components/templates/SignUp/SignUp';
import { GlobalStyle } from '../assets/styles/GlobalStyle';
import { theme } from '../assets/styles/theme';
import MainContainerApp from '../components/templates/MainContainer/MainContainer';
import AboutUs from '../components/templates/AboutUs/AboutUs';
import Contact from '../components/templates/Contact/Contact';
import LogIn from '../components/templates/Login/LogIn';
import AdminDashboard from '../components/templates/Admin_Dashboard/Admin_Dashboard';
import UserDashboard from '../components/templates/User_Dashboard/UserDashboard';
import { Context } from '../providers/GeneralProvider';
import ClientsOrProjects from '../components/templates/Admin_ClientsOrProjects/ClientsOrProjects';
import Settings from '../components/templates/Settings/Settings';
import NewClient from '../components/templates/Admin_NewClient/NewClient';
import Projects from '../components/templates/User_Projects/Projects';
import VerifyEmail from '../components/molecules/VerifyEmail/VerifyEmail';
import ForgotPassword from '../components/molecules/ForgotPassword/ForgotPassword';
import ResetPassword from '../components/molecules/ResetPassword/ResetPassword';
import Statistics from '../components/templates/Admin_Statistics/Statistics';
import ClientDetails from '../components/organisms/ClientDetails/ClientDetails';
import ProjectDetail from '../components/organisms/ProjectDetails/ProjectDetails';
import Messages from '../components/templates/Messages/Messages';
import NewProject from '../components/templates/Admin_NewProject/NewProject';
import Impressum from '../components/templates/Impressum/Impressum';
import { useAuth } from '../hooks/useAuth';
import useError from '../hooks/useError';
import { Faq } from '../components/molecules/FAQs/FAQs';
import EditProject from '../components/organisms/EditProject/EditProject';
import Page404 from '../components/templates/Page404/Page404';

function App(): JSX.Element {
  const { userData, setMessages, setUserData, setClientsGlobal } = useContext(Context);
  const [messageDisplay, setMessageDisplayed] = useState(false);
  const [listening, setListening] = useState(false);
  const navigate = useNavigate();
  const { handleLogout } = useAuth();
  const { token, role } = userData;
  const { handleError } = useError();

  useEffect(() => {
    let interval: any;
    if (userData.token) {
      // When entering the website, check whether the token's time has expired
      if (Date.now() > userData.exp) {
        handleLogout();
        setListening(false);
        setMessageDisplayed(false);
      }
      // If token exist check whether the token is close to expiration ( < 30s )
      if (userData.token) {
        interval = setInterval(() => {
          // If yes then Display Message About It
          if (+userData.exp - Date.now() < 30000 && !messageDisplay) {
            handleError('For Your Safety We Will Logout You in a 30 seconds :)');
            setMessageDisplayed(true);
          }
          // If Token expired Clear UserData and Logout User/Admin
          if (Date.now() > userData.exp) {
            handleLogout();
            setListening(false);
            handleError('We Logged Out You For Your Security :)', true);
            clearInterval(interval);
            setMessageDisplayed(false);
          }
          // console.log('Left', ((userData.exp - Date.now()) / 1000).toFixed(0), 's To Logout');
        }, 5000);
      } else {
        clearInterval(interval);
      }
    }
    return () => clearInterval(interval);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [userData.exp]);

  // Fetching clientsGlobal from Freelancer
  const fetchClients = async () => {
    try {
      const res = await fetch(`${process.env.REACT_APP_BACKEND}/user/freelancer`, {
        method: 'GET',
        headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${userData?.token}`
        }
      });
      const resJSON = await res.json();
      // console.log(resJSON);
      if (res.status === 200) {
        setClientsGlobal(resJSON);
      } else {
        handleError(resJSON.message);
      }
    } catch (error: any) {
      console.log('FETCHING ERROR', error);
      handleError();
    }
  };
  // Fetching Freelancers from Client
  const fetchClientsForClient = async () => {
    try {
      const res = await fetch(`${process.env.REACT_APP_BACKEND}/user/freelancers`, {
        method: 'GET',
        headers: {
          'Content-type': 'application/json',
          Authorization: `Bearer ${userData?.token}`
        }
      });
      const resJSON = await res.json();
      // console.log(resJSON);
      if (res.status === 200) {
        setClientsGlobal(resJSON);
      } else {
        handleError(resJSON.message);
      }
    } catch (error: any) {
      console.log('FETCHING ERROR', error);
      handleError();
    }
  };
  // Fetching clientsGlobal (Freelancer or Client)
  useEffect(() => {
    if (token && role === 'Freelancer') {
      fetchClients();
    }
    if (token && role === 'Client') {
      fetchClientsForClient();
    }
  }, [token]);

  useEffect(() => {
    const connectSSE = async () => {
      if (token) {
        if (!listening) {
          console.log('I try listening SSE...');
          const events = new EventSource(`${process.env.REACT_APP_BACKEND}/events/${token}`);

          events.onmessage = (event) => {
            const parsedData = JSON.parse(event.data);
            console.log('Parsed', parsedData);
            if (parsedData.text === 'stopSSEEventsNow') {
              events.close();
              console.log('I closed the connection Here');
            }
            setMessages((messagesItems) => messagesItems.concat(parsedData));
          };

          events.onerror = (e) => {
            console.log('SSE ERROR', e);
          };

          events.onopen = () => {
            console.log('Connection works ...');
          };

          setListening(true);
        }
        console.log('I do Nothing');
      } else {
        setMessages([]);
        setListening(false);
      }
    };
    connectSSE();
  }, [listening, token]);

  const handleGoogleLogin = async (code: any) => {
    try {
      const res = await fetch(`${process.env.REACT_APP_BACKEND}/googleLogin`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
          // Authorization: 'Bearer ' + userData?.token, // IN FUTURE TO AUTHORIZATION
        },
        body: JSON.stringify({ code })
      });
      const resJSON = await res.json();
      if (res.status === 200) {
        setUserData(resJSON);
        navigate('/');
        handleError('You are correctly logged in', true);
      } else {
        handleError(resJSON.message, res.status === 200);
      }
    } catch (error: any) {
      console.log('FETCHING ERROR', error);
      handleError();
    }
  };

  useEffect(() => {
    const urlParams = queryString.parse(window.location.search);

    // Finded another solution? TODO - check this, maybe better avoid installing a library
    // const urlParams = new URLSearchParams(window.location.search);
    // const clientCode = urlParams.get('code');

    if (urlParams.error) {
      const { error }: any = urlParams;
      console.log(`An error occurred: ${error}`);
      handleError('Something went wrong with Google Login, try later again');
    }
    if (urlParams.code) {
      console.log(`The code is: ${urlParams.code}`);
      handleGoogleLogin(urlParams.code);
    }
  }, [window.location.search]);

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      <MainContainerApp>
        <Routes>
          {token && role === 'Freelancer' ? (
            <>
              <Route path="/" element={<AdminDashboard />} />
              <Route path="/clients" element={<ClientsOrProjects />} />
              <Route path="/client/:clientId" element={<ClientDetails />} />
              <Route path="/project/:projectId" element={<ProjectDetail />} />
              <Route path="/editProject/:projectId" element={<EditProject />} />
              <Route path="/settings" element={<Settings />} />
              <Route path="/statistics" element={<Statistics />} />
              <Route path="/messages" element={<Messages />} />
              <Route path="/newClient" element={<NewClient />} />
              <Route path="/newProject/:clientId" element={<NewProject />} />
            </>
          ) : token && role === 'Client' ? (
            <>
              <Route path="/" element={<UserDashboard />} />
              <Route path="/projects" element={<Projects />} />
              <Route path="/project/:projectId" element={<ProjectDetail />} />
              <Route path="/client/:clientId" element={<ClientDetails />} />
              <Route path="/messages" element={<Messages />} />
              <Route path="/settings" element={<Settings />} />
            </>
          ) : (
            <>
              <Route path="/" element={<HomePage />} />
              <Route path="/signup" element={<SignUp />} />
              <Route path="/login" element={<LogIn />} />
              <Route path="/verifyEmail/:token" element={<VerifyEmail />} />
              <Route path="/forgotPassword/" element={<ForgotPassword />} />
              <Route path="/forgotPassword/:token" element={<ResetPassword />} />
            </>
          )}
          <Route path="/contact" element={<Contact />} />
          <Route path="/services" element={<Services />} />
          <Route path="/aboutUs" element={<AboutUs />} />
          <Route path="/impressum" element={<Impressum />} />
          <Route path="/faq" element={<Faq />} />
          <Route path="/*" element={<Page404 />} />
        </Routes>
      </MainContainerApp>
    </ThemeProvider>
  );
}

export default App;
