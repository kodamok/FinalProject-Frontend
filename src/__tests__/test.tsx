import * as React from 'react';
import '../helpers/matchMedia';
import { fireEvent, screen } from '@testing-library/react';
import i18next from 'i18next';
import HttpApi from 'i18next-http-backend';
import LanguageDetector from 'i18next-browser-languagedetector';
import { initReactI18next } from 'react-i18next';
import { renderWithThemeProvider } from '../helpers/renderWithThemeProvider';
import HomePage from '../components/templates/HomePage/HomePage';
import App from '../view/App';
import ClientsOrProjects from '../components/templates/Admin_ClientsOrProjects/ClientsOrProjects';
import Login from '../components/templates/Login/LogIn';

beforeEach(() => {
  i18next
    .use(HttpApi)
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
      supportedLngs: ['en', 'de'],
      fallbacks: true,
      fallbackLng: 'en',
      whitelist: 'en',
      debug: false,
      // Options for language detector
      detection: {
        order: [
          'cookie',
          'localStorage',
          'sessionStorage',
          'navigator',
          'htmlTag',
          'querystring',
          'path',
          'subdomain'
        ],
        lookupQuerystring: 'lng',
        lookupCookie: 'i18n',
        lookupLocalStorage: 'i18nextLng',
        caches: ['localStorage', 'cookie']
      },
      backend: {
        loadPath: '/assets/locales/{{lng}}/translation.json'
      },
      react: {
        useSuspense: false
      }
    } as any);
});

describe('Test Snapshots', () => {
  it('Simple Test', () => {
    expect(2 + 2).toBe(4);
  });
  it('<HomePage />', () => {
    const comp = renderWithThemeProvider(<HomePage />);
    expect(comp.container.firstChild).toMatchSnapshot();
  });
  it('<App />', () => {
    const comp = renderWithThemeProvider(<App />);
    expect(comp.container.firstChild).toMatchSnapshot();
  });
  it('<ClientsOrProjects />', () => {
    const comp = renderWithThemeProvider(<ClientsOrProjects />);
    expect(comp.container.firstChild).toMatchSnapshot();
  });
  it('<Login />', () => {
    const comp = renderWithThemeProvider(<Login />);
    fireEvent.click(screen.getByText('Login'));
    setTimeout(() => {
      screen.getByText('Oops');
    }, 1000);
    expect(comp.container.firstChild).toMatchSnapshot();
  });
});

// const TYPE_NUMBER = 'number';
//
// export function add(a: any, b: any) {
//   // First check if a and b exist (zero is an allowed value), then check their types are correct
//   // eslint-disable-next-line valid-typeof
//   if ((a || a === 0) && (b || b === 0) && typeof a === TYPE_NUMBER && typeof b === TYPE_NUMBER) {
//     return a + b;
//   }
//   // If there's anything wrong with the input data, return false
//   return false;
// }
//
// const tests = [
//   // First check the happy path - the normal inputs expected to get an output
//   { inputs: [0, 0], expected: 0 },
//   { inputs: [0, 1], expected: 1 },
//   { inputs: [1, 1], expected: 2 },
//   { inputs: [1000, 1000], expected: 2000 },
//   // Now check the sad path - both extreme and strange inputs and what output they should give
//   { inputs: [100000000000000000, 100000000000000000], expected: 200000000000000000 },
//   { inputs: [false, false], expected: false },
//   { inputs: [null, null], expected: false },
//   { inputs: [undefined, undefined], expected: false },
//   { inputs: [NaN, NaN], expected: false },
//   { inputs: [false, false], expected: false }
// ];
//
// // Using forEach, lets loop over each test configuration, create a new Jest test for each of them,
// // and load in the test configuration into that test
// tests.forEach((config: any) => {
//   test(`Adding ${config.inputs[0]} and ${config.inputs[1]} should equal ${config.expected}`, () => {
//     expect(add.apply(this, config.inputs)).toEqual(config.expected);
//   });
// });
