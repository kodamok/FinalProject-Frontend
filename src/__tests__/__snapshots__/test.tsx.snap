// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshots <App /> 1`] = ``;

exports[`Test Snapshots <ClientsOrProjects /> 1`] = ``;

exports[`Test Snapshots <HomePage /> 1`] = ``;

exports[`Test Snapshots <HomePage /> render 1`] = ``;

exports[`Test Snapshots <Login /> 1`] = ``;
